### PR TITLE
Perceiver

### PR DESCRIPTION
**Pull request by kaiyuanmifen**
Perceiver + embedding based transformer

### CODE REVIEW

File: ml4h/models/perceiver_blocks.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    # )({
+    #     'query': latents,
+    #     'key': latents,
+    #     'value': latents,
+    #     # No mask
+    # })
+    # attention = tf.keras.layers.Dropout(rate=dropout)(attention)
+    # latents = tf.keras.layers.LayerNormalization(
+    #     epsilon=1e-6,
+    # )(latents + attention)
```
Comment (lucidtronix) [inline ml4h/models/perceiver_blocks.py:213]: remove this or provide alternate control flow?

**Review by lucidtronix — CHANGES_REQUESTED** [review]
This is great @kaiyuanmifen !  I'm still reviewing the model blocks, but I did a pass just doing small changes. And we should look into why the tests are failing

**Comment by lucidtronix** [comment]
@kaiyuanmifen I think the serialization issue is because the custom classes you've added need to also be added into this function: `_get_custom_objects` line 304 of `model_factory.py` https://github.com/broadinstitute/ml4h/blob/Dianbo/ml4h/models/model_factory.py#L304

Can you try adding all the newly defined custom objects there and then seeing if serialization works.

**Comment by kaiyuanmifen** [comment]
Dear Sam

Yes, there was a problem running the serializaiton of the new part and
save the model, which I have not solved. The training runs ok.

**Comment by kaiyuanmifen** [comment]
fixing saving and inference

**Review by lucidtronix — APPROVED** [review]

**Pull request merged**